fuel-relayer/README.md

```diff
@@ -0,0 +1,116 @@
+
+### Syncing flow
+
+Tricky thing that can happen not so often is in initial synchronization. When do we start log subscription and when do we stop importing log from block. Initial sync is done as:
+1. sync from HardCoddedContractCreatingBlock to (BestEthBlock-EthFinalitySlider)
+2. sync overlap from LastIncludedEthBlock to BestEthBlock and save them in dequeue.
+3. Start listening to eth events
+4. Check if our LastIncludedEthBlock is same as BestEthBlock.
+  If not the same, stop listening to events and do 2,3,4 steps again.
+7. Continue to active listen on eth events. and prune(commit to db) dequeue for older finalized events
+
+To have validity of ValidatorSet we need to have logs from ethereum to reconstruct the set. So It does not matter for us if we start syncing to fuel network if we are not connected to eth client. And in case if fuel-bft ask for validator set of not yet synced relayer, we are going to return error. We probably need to have notification when relayer gets fully synced.
+
+On active sync when initial sync is finished we are watching new eth events and new chained block:
+For log watch we are doing:
+1. Receive log event:
+    1. if it is removal, append them into pending_removal vec
+    2. if it is new event. Apply pending_removal_vec and add new event to dequeue.
+2. On new fuel block, check finality and commit changed to db.
+     1. just to be sure introduce check of events by calling eth block logs. This is a double check just to be sure that we match with that is inside contract.
+
+For passive sync we are using ethereum pubsub protocol to get `logs` event: https://geth.ethereum.org/docs/rpc/pubsub . Example of eth log from log subscription:
+* "address":0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,
+* "topics":[0x00,0x00,0x00],
+* "data":"Bytes",
+* "block_hash":Some(Hash)),
+* "block_number":Some(14069822),
+* "transaction_hash":Some(0xcf7baaa6e6cd2d863807a3b8d3168f8d2699ad043e3ac249e82887fda6c4ef55),
+* "transaction_index":Some(130),
+* "log_index":Some(172),
+* "transaction_log_index":"None",
+* "log_type":"None",
+* "removed":"Some(false)"
+
+### Database:
+
+We are adding three colums:
+* Validator set diff: mapping of fuel_block and hashmap of diffs for finalized new stakes.
+* Validator set: mapping of validator address to its current stakes.
+* Deposits: mapping of deposit_nonce to token deposits.
+
+## Block Commitments
+
+When new fuel block is made it needs to be commited to ethereum network.
+
+There is timeframe of few eth blocks where current leader can broadcast block to ethereum network. In case that he do it in time he will receive additional reward.
+
+In case that leader does not commit fuel block next leader will have ability to do it for him as bundle block commit, in that case bonus reward will go to second leader that has spend additional gas to push this change to ethereum network.
```

Comment:
```suggestion
In case that leader does not commit fuel block next leader will have ability to do it for him as bundle block commit.
```

again, these reward schemes you came up with were not agreed on or discussed in the specs.

Reply:
The specs don't yet cover these cases. As such, preference should be to do the simplest thing possible at first (i.e. nothing), and document that it may change in the future.

Reply:
@adlerjohn @Voxelot this felt like a good response to the first comment. As already demonstrated with byte price and key rotation those were not the simplest solution for the problem but addons to already functional things.
I am not sure what I am missing here as a reward system seems like a pretty simple, logical and needed thing to do, and from both of you, I didn't get the reasoning why do you want it removed or put something else in place of it?